[BUG] javax.net.ssl.SSLHandshakeException when using `verify(false)`

**Describe the bug**

According to the documentation if we want disable SSL check, we need use the `verify(false)` method present in `SslConfig` class. However, this causes the following error:

```
Caused by: javax.net.ssl.SSLHandshakeException: No subject alternative DNS name matching vault.domain.local found.
```

**To Reproduce**

Use this configuration:

```java
final VaultConfig vaultConfig = new VaultConfig()
                .address("https://vault.domain.local")
                .token("mock_token")
                .sslConfig(new SslConfig().verify(false))
                .engineVersion(1)
                .build();
```

--- a/build.gradle
+++ b/build.gradle
@@ -19,10 +19,10 @@ repositories {
 
 dependencies {
     testImplementation('junit:junit:4.13.2')
-    testImplementation('org.mockito:mockito-core:5.2.0')
-    testImplementation('org.testcontainers:testcontainers:1.17.6')
-    testImplementation('org.eclipse.jetty:jetty-server:11.0.14')
-    testImplementation('org.slf4j:slf4j-api:2.0.5')
+    testImplementation('org.mockito:mockito-core:5.4.0')
+    testImplementation('org.testcontainers:testcontainers:1.19.1')
+    testImplementation('org.eclipse.jetty:jetty-server:11.0.15')
+    testImplementation('org.slf4j:slf4j-api:2.0.7')
     testImplementation('org.bouncycastle:bcprov-jdk15on:1.70')
     testImplementation('org.bouncycastle:bcpkix-jdk15on:1.70')
     testImplementation('org.apache.commons:commons-io:1.3.2')

--- a/src/main/java/io/github/jopenlibs/vault/rest/Rest.java
+++ b/src/main/java/io/github/jopenlibs/vault/rest/Rest.java
@@ -4,6 +4,7 @@ import io.github.jopenlibs.vault.SslConfig;
 import io.github.jopenlibs.vault.VaultConfig;
 import java.io.IOException;
 import java.io.UnsupportedEncodingException;
+import java.net.Socket;
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.URLEncoder;
@@ -27,8 +28,9 @@ import java.util.Optional;
 import java.util.StringJoiner;
 import java.util.TreeMap;
 import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
 import javax.net.ssl.TrustManager;
-import javax.net.ssl.X509TrustManager;
+import javax.net.ssl.X509ExtendedTrustManager;
 
 /**
  * <p>A simple client for issuing HTTP requests.  Supports the HTTP verbs:</p>
@@ -77,7 +79,27 @@ public class Rest {
     static {
         try {
             DISABLED_SSL_CONTEXT = SSLContext.getInstance("TLSv1.2");
-            DISABLED_SSL_CONTEXT.init(null, new TrustManager[]{new X509TrustManager() {
+            DISABLED_SSL_CONTEXT.init(null, new TrustManager[]{new X509ExtendedTrustManager() {
+                @Override
+                public void checkClientTrusted(X509Certificate[] chain, String authType,
+                        Socket socket) throws CertificateException {
+                }
+
+                @Override
+                public void checkServerTrusted(X509Certificate[] chain, String authType,
+                        Socket socket) throws CertificateException {
+                }
+
+                @Override
+                public void checkClientTrusted(X509Certificate[] chain, String authType,
+                        SSLEngine engine) throws CertificateException {
+                }
+
+                @Override
+                public void checkServerTrusted(X509Certificate[] chain, String authType,
+                        SSLEngine engine) throws CertificateException {
+                }
+
                 @Override
                 public void checkClientTrusted(final X509Certificate[] x509Certificates,
                         final String s) throws CertificateException {

